tools/run_tests/filter_pull_request_tests.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python2.7
+# Copyright 2015, Google Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Filter out tests based on file differences compared to merge target branch"""
+
+from subprocess import call, check_output
+
+# Whitelist for all tests
+# If whitelist item should only trigger some tests, the item should be
+# added to this list and the trigger list of tests that should be run
+starts_with_whitelist = ['templates/',
+                         'doc/',
+                         'examples/',
+                         'summerofcode/',
+                         'src/cpp',
+                         'src/csharp',
+                         'src/node',
+                         'src/objective-c',
+                         'src/php',
+                         'src/python',
+                         'src/ruby',
+                         'test/core',
+                         'test/cpp',
+                         'test/distrib/cpp',
+                         'test/distrib/csharp',
+                         'test/distrib/node',
+                         'test/distrib/php',
+                         'test/distrib/python',
+                         'test/distrib/ruby']
+				   
+ends_with_whitelist = ['README.md',
```

Comment:
nit: would it make sense to use regular expressions to match the path instead of starts_with and ends_with.

`'^src/cpp'` and `'README.md$'` seem quite self-explanatory.